packages/core-typings/src/IRoom.ts

```diff
@@ -95,8 +97,29 @@ export interface IRoom extends IRocketChatRecord {
 	customFields?: Record<string, any>;
 
 	usersWaitingForE2EKeys?: { userId: IUser['_id']; ts: Date }[];
+
+	sidepanel?: {
+		items: [SidepanelItem, SidepanelItem?];
+	};
 }
 
+export const isSidepanelItem = (item: any): item is SidepanelItem => {
+	return sidepanelItemValues.includes(item);
+};
+
+export const isValidSidepanel = (sidepanel: IRoom['sidepanel']) => {
+	if (!sidepanel?.items) {
+		return false;
+	}
+	return (
+		Array.isArray(sidepanel.items) &&
+		sidepanel.items.length &&
+		sidepanel.items.every(isSidepanelItem) &&
+		sidepanel.items.length <= 2 &&
+		sidepanel.items.length === new Set(sidepanel.items).size
```

Comment:
```suggestion
		sidepanel.items.length === new Set(sidepanel.items).size
```
I believe the second condition makes us not need the first one

Reply:
noooot really, the second is to ensure there are no duplicates on the array - if there are, the length of the `new Set` will be different, because it will remove the duplicate.
the first oder condition is to ensure it has max 2 items :)

Reply:
hmmmm but if there are no duplicates and all items are valid, it should be safe to say the array has max 2 elements. shouldn't it? :thinking:

Reply:
hammm you're right!